src/main/java/org/checkstyle/autofix/CheckstyleRecipeRegistry.java

```diff
@@ -65,18 +67,49 @@ private CheckstyleRecipeRegistry() {
      * @return a list of generated Recipe objects
      */
     public static List<Recipe> getRecipes(List<CheckstyleViolation> violations,
-                                          Map<CheckstyleCheck, CheckConfiguration> config) {
+                                          Map<CheckstyleConfigModule, CheckConfiguration> config) {
         return violations.stream()
-                .collect(Collectors.groupingBy(CheckstyleViolation::getSource))
+                .collect(Collectors.groupingBy(CheckstyleViolation::getCheckId))
                 .entrySet()
                 .stream()
                 .map(entry -> {
-                    return createRecipe(entry.getValue(), config.get(entry.getKey()));
+                    final CheckConfiguration configuration =
+                            findMatchingConfiguration(entry.getKey(), config);
+                    return createRecipe(entry.getValue(), configuration);
                 })
                 .filter(Objects::nonNull)
                 .collect(Collectors.toList());
     }
 
+    private static CheckConfiguration findMatchingConfiguration(String source,
+                                           Map<CheckstyleConfigModule, CheckConfiguration> config) {
+        return config.entrySet().stream()
+                .filter(configEntry -> matchesSource(configEntry.getKey(), source))
+                .map(Map.Entry::getValue)
+                .findFirst()
+                .orElse(null);
```

Comment:
could you return Optional and filter out absent values before trying to create recipe after calling `findMatchingConfiguration`